protolog/loglistener.go

```diff
@@ -144,86 +132,123 @@ func (ll *LogListener) Shutdown() {
 	close(ll.logEntriesRecieved)
 }
 
+func (ll *LogListener) processConnection(conn net.Conn) {
+	if ll.config.JsonMode {
+		buffer := Buffer{conn, make([]byte, 0)}
```

Comment:
Use the `NewBuffer` function? I think we should also default allocation to a reasonable size, i.e. something around the 90% percentile. My thinking is that while this will use more memory than needed sometimes, it will reduce the number of memory allocations. As this is likely a long running process and we keep the slive for reuse, it will grow to this over time anyway, so doing it up front will be more performant.